homeassistant/components/smlight/coordinator.py

```diff
@@ -85,12 +91,13 @@ async def _async_setup(self) -> None:
 
     def update_setting(self, setting: Settings, value: bool | int) -> None:
         """Update the sensor value from event."""
+
         prop = SettingsProp[setting.name].value
         setattr(self.data.sensors, prop, value)
 
         self.async_set_updated_data(self.data)
 
-    async def _async_update_data(self) -> SmData:
+    async def _async_update_data(self) -> SmData | SmFwData:
```

Comment:
Why do you change this type? If you share a lot of logic I think creating a base coordinator is better. Withings has some of these

Reply:
ok I see, yes base coordinator here is cleaner.